contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java

```diff
@@ -150,14 +150,11 @@ public AbstractKinesisInputOperator()
     windowDataManager = new FSWindowDataManager();
     currentWindowRecoveryState = new HashMap<String, KinesisPair<String, Integer>>();
   }
+
   /**
-   * Derived class has to implement this method, so that it knows what type of message it is going to send to Malhar.
-   * It converts a ByteBuffer message into a Tuple. A Tuple can be of any type (derived from Java Object) that
-   * operator user intends to.
-   *
-   * @param rc Record to convert into tuple
+   * Any concrete class derived from AbstractKinesisInputOperator to implement this method to emit tuples to an output port.
    */
-  public abstract T getTuple(Record rc);
+  protected abstract void emitTuple(Record rc);
```

Comment:
Why getTuple(..) method deleted ? This comes under backward compatible.
I would suggest define the emitTuple method as below:
```java
protected T emitTuple(Pair<String, Record> data)
{
  return getTuple(data.getSecond());
}
```